device/iot-device-samples/multiplexing-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/ClientManagerBase.java

```diff
@@ -0,0 +1,166 @@
+package samples.com.microsoft.azure.sdk.iot;
```

Comment:
Given this is a sample and that the task is complex, it is going to need way more code comments.

I'd like to see high level approach, and then for each method how it fits into the larger picture.

Reply:
good point. will add a lot more comments.